cylc/flow/workflow_files.py

```diff
@@ -1320,6 +1320,45 @@ def infer_latest_run(path: Path) -> Tuple[Path, Path]:
     return (path, reg)
 
 
+def check_nested_run_path_base(path_: Union[Path, str]) -> None:
```

Comment:
Note: would probably be best to merge #4448 before tackling

Reply:
Further development (noting down what we discussed in Teams just now): as I'm now stripping out the children checking in #4448, the problem becomes _O(n)_ (linear with the number of nested parents only), so it would be fine to have the install dir checking in its own function.